src/test/java/de/tum/in/www1/artemis/StudentExamIntegrationTest.java

```diff
@@ -638,8 +638,13 @@ public void testAssessUnsubmittedStudentExams() throws Exception {
                 HttpStatus.OK, null);
         database.changeUser("instructor1");
         Set<StudentExam> unsubmittedStudentExams = studentExamRepository.findAllUnsubmittedWithExercisesByExamId(exam2.getId());
-        Map<User, List<Exercise>> exercisesOfUser = unsubmittedStudentExams.stream().collect(Collectors.toMap(StudentExam::getUser, studentExam -> studentExam.getExercises()
-                .stream().filter(exercise -> exercise instanceof ModelingExercise || exercise instanceof TextExercise).collect(Collectors.toList())));
+        Map<User, List<Exercise>> exercisesOfUser = unsubmittedStudentExams.stream()
+                .collect(
+                        Collectors
+                                .toMap(StudentExam::getUser,
+                                        studentExam -> studentExam.getExercises().stream().filter(exercise -> exercise instanceof ModelingExercise
+                                                || exercise instanceof TextExercise || exercise instanceof FileUploadExercise || exercise instanceof ProgrammingExercise)
+                                                .collect(Collectors.toList())));
```

Comment:
Here the production code is just duplicated multiple times in the test cases. Can you find a better solution for this? I don't know how these test exams are created, but maybe you can just create this Map via `Map.of()`?